app/controllers/UndeliverableEmailController.scala

```diff
@@ -43,22 +44,55 @@ class UndeliverableEmailController @Inject() (
       request.body.extractEori match {
 
         case Some(eori) =>
-          emailRepository
-            .findAndUpdate(eori, request.body)
-            .flatMap {
-              case Some(record) =>
-                auditingService.auditBouncedEmail(request.body)
-                updateSub22(request.body, record, eori).map(_ => NoContent)
-              case _            => Future.successful(NotFound)
-            }
-            .recover { case err =>
-              log.error(s"Failed to mark email as undeliverable: ${err.getMessage}"); InternalServerError
+          val result = for {
```

Comment:
You could simplify this by avoiding the `for` block and just pattern matching directly on the `Future[Option[NotificationEmail]]`. Here's a possible refactor:

```suggestion
          val result: Future[Result] = emailRepository.get(eori).flatMap {
  case Some(notifEmail) if notifEmail.address.nonEmpty =>
    checkEmailAddressAndProcessUndeliverableInfo(request, eori, notifEmail.address)
  case _ =>
    Future.successful(NotFound)
}
```

Reply:
Done